### DISCUSSION TITLE
Suggestion: highlight matching words/keywords in issue search results

### DISCUSSION THREAD

**Discussion by hsandt** [community/community#102767] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

I often find myself typing keywords to search for existing issues on a topic. Since the search is so broad, going as far as finding matching words in every comment (which is very useful btw), I find many matching issues but I want to prioritize those that actually contain the searched words in their *title*.

So I must then press Ctrl+F and enter the same words again (this time one by one since they may not be in the same order in the titles, and browser search generally looks for exact sentence) to actually find them in the issue titles.

It would be nice if the matching words were highlighted in the filter results issue titles, as in other apps such as Thunderbird, to spot them quickly.

If it's bothering some people, it may become an option in the Settings (maybe Appearance of Accessibility - I didn't post this in Accessibility though because it's more like an extra feature for me, but it could also help people who have difficulties reading many titles fast - and the highlight color itself could also be in Accessibility).

It would be more difficult for non-exact matches though: similarly to saerch engines, we'd have to decide whether or not to highlight similar, but non-exact matches (assuming GitHub issue search does that). For now, I only care about exact matches.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by Olayiwolaaa** [comment]
Your suggestion to have matching words highlighted in the filter results issue titles on GitHub is a practical and user-friendly feature that could enhance the search experience for users like yourself. Highlighting the keywords within the titles would indeed streamline the process of identifying relevant issues without the need to manually search within each issue.

Integrating this feature as an option in the Settings, specifically under Accessibility or Appearance, is a thoughtful approach to ensure flexibility for users who may prefer or benefit from such highlighting. Offering customizable highlight colors, especially considering accessibility needs, demonstrates a commitment to inclusivity.

Addressing non-exact matches could be a nuanced consideration. While exact matches are your primary concern, it's worth exploring how GitHub's search engine handles similar, but non-exact matches, and whether highlighting such matches would be beneficial or potentially overwhelming for users.

Overall, your suggestion aligns with improving user experience by optimizing the efficiency of issue searching on GitHub, and integrating it as an optional feature reflects a user-centric design approach.
